### PR TITLE
Fix: Display of all object names in schema summary of analyze report 

### PR DESCRIPTION
**Pull request by priyanshi-yb**
Description:

1. Changed the type of Object set in the SummaryMap from a Map to an Array so also list the multiple object names which is possible for INDEX, TRIGGER, and POLICY objects. 
For the counts to be proper, fixed some of the invalidCount for these object types to set the object as invalid with proper name.
 https://yugabyte.atlassian.net/browse/DB-11409
3. Fixed a bug where `InvalidCount` for Indexes for getting set for all the indexes as in `reportCreateIndexStorageParamenter()` func the `invalidCount` was set for all indexes instead it should be done inside check condition of storage parameter. 
2. Updated the Assessment report to not put MigrationCaveats section in case there are no caveats.
3. Modified the `IssueType` for `Foreign table` and `Alter partition table to add PK` issues to `migration_caveats`.

### CODE REVIEW

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 1]
```diff
@@ -993,9 +993,12 @@ func addMigrationCaveatsToAssessmentReport(unsupportedDataTypesForLiveMigration
 			}
 			migrationCaveats = append(migrationCaveats, UnsupportedFeature{"Unsupported Data Types for Live Migration", columns, false, "", UNSUPPORTED_DATATYPES_FOR_LIVE_MIGRATION_ISSUE})
 		}
-
-		if len(migrationCaveats) > 0 {
-			assessmentReport.MigrationCaveats = migrationCaveats
+		for _, caveat := range migrationCaveats {
```
Comment (makalaaneesh) [inline yb-voyager/cmd/assessMigrationCommand.go:996]: pls explain what is being done here and add TODO for proper solution.

File: yb-voyager/cmd/analyzeSchema.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ import (
 type summaryInfo struct {
 	totalCount   int
 	invalidCount map[string]bool
-	objSet       map[string]bool
+	objSet       []string
```
Comment (makalaaneesh) [inline yb-voyager/cmd/analyzeSchema.go:46]: Add a todo here as well. Ultimately we want to still use a set, not an array

**Review by makalaaneesh — APPROVED** [review]
LGTM with a couple of comments

File: yb-voyager/cmd/analyzeSchema.go (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go]: https://yugabyte.atlassian.net/browse/DB-12602

**Pull request merged**